integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/management/RefreshCarbonTableCommand.scala

```diff
@@ -0,0 +1,264 @@
+
+import java.util
+
+import scala.collection.JavaConverters._
+
+import org.apache.spark.sql._
+import org.apache.spark.sql.execution.command.MetadataCommand
+import org.apache.spark.sql.util.CarbonException
+
+import org.apache.carbondata.common.logging.{LogService, LogServiceFactory}
+import org.apache.carbondata.core.constants.CarbonCommonConstants
+import org.apache.carbondata.core.datastore.impl.FileFactory
+import org.apache.carbondata.core.locks.{ICarbonLock, LockUsage}
+import org.apache.carbondata.core.metadata.{AbsoluteTableIdentifier, CarbonTableIdentifier}
+import org.apache.carbondata.core.metadata.converter.ThriftWrapperSchemaConverterImpl
+import org.apache.carbondata.core.metadata.schema.table.{DataMapSchema, TableInfo}
+import org.apache.carbondata.core.metadata.schema.table.column.ColumnSchema
+import org.apache.carbondata.core.util.CarbonProperties
+import org.apache.carbondata.core.util.path.CarbonStorePath
+import org.apache.carbondata.events.{CreateTablePreExecutionEvent, OperationContext, OperationListenerBus, RefreshTablePostExecutionEvent, RefreshTablePreExecutionEvent}
+import org.apache.carbondata.spark.util.CommonUtil
+
+/**
+ * Command to register carbon table from existing carbon table data
+ */
+case class RefreshCarbonTableCommand(
+    dbName: Option[String],
+    tableName: String)
+  extends MetadataCommand {
+  val LOGGER: LogService =
+    LogServiceFactory.getLogService(this.getClass.getName)
+
+  override def processMetadata(sparkSession: SparkSession): Seq[Row] = {
+    val metaStore = CarbonEnv.getInstance(sparkSession).carbonMetastore
+    val databaseName = GetDB.getDatabaseName(dbName, sparkSession)
+    val databaseLocation = GetDB.getDatabaseLocation(databaseName, sparkSession,
+      CarbonProperties.getStorePath)
+    // Steps
+    // 1. get table path
+    // 2. perform the below steps
+    // 2.1 check if the table already register with hive then ignore and continue with the next
+    // schema
+    // 2.2 register the table with the hive check if the table being registered has aggregate table
+    // then do the below steps
+    // 2.2.1 validate that all the aggregate tables are copied at the store location.
+    // 2.2.2 Register the aggregate tables
+    val locksToBeAcquired = List(LockUsage.METADATA_LOCK, LockUsage.DROP_TABLE_LOCK)
+    var locks = List.empty[ICarbonLock]
+    try {
+      val tablePath = databaseLocation + CarbonCommonConstants.FILE_SEPARATOR + tableName
+      val absoluteTableIdentifier = AbsoluteTableIdentifier.from(tablePath, databaseName, tableName)
+      locks = CommonUtil.acquireLock(locksToBeAcquired, absoluteTableIdentifier)
+      // 2.1 check if the table already register with hive then ignore and continue with the next
+      // schema
+      if (!sparkSession.sessionState.catalog.listTables(databaseName)
+        .exists(_.table.equalsIgnoreCase(tableName))) {
+        val carbonTablePath = CarbonStorePath.getCarbonTablePath(absoluteTableIdentifier)
```

Comment:
Please move reading of TableInfo using tablePath to the CarbonMetaStore

Reply:
moved to CarbonMetaStore.getWrapperTableInfo(absoluteTableIdentifier)